src/relay.cpp

```diff
@@ -358,82 +373,148 @@ void prepare_relay_config(relay_config &interface_config, int local_sock, int fi
     }
     freeifaddrs(ifa); 
     
+    char ipv6_str[INET6_ADDRSTRLEN] = {};
     if(!IN6_IS_ADDR_LINKLOCAL(&non_link_local.sin6_addr)) {
         interface_config.link_address = non_link_local;
+        inet_ntop(AF_INET6, &non_link_local.sin6_addr, ipv6_str, INET6_ADDRSTRLEN);
```

Comment:
lines 379,380 & 384,385 can be moved after the if condition.

Reply:
Good catch, will apply in next diff.

Reply:
Applied in latest diff.